src/test/java/com/powsybl/openloadflow/ac/SecondaryVoltageControlTest.java

```diff
@@ -362,4 +367,65 @@ void testNotPlausibleTargetV() {
         LoadFlowResult result = loadFlowRunner.run(network, parameters);
         assertEquals(LoadFlowResult.ComponentResult.Status.FAILED, result.getComponentResults().get(0).getStatus());
     }
+
+    @Test
+    void testWithGeneratorRemoteVoltage() {
+        // move generator 6 to bus 5 but keep voltage control on bus 6
+        network.getGenerator("B6-G").remove();
+        var g5 = network.getVoltageLevel("VL5").newGenerator()
+                .setId("B5-G")
+                .setBus("B5")
+                .setTargetP(0)
+                .setMinP(-9999)
+                .setMaxP(9999)
+                .setTargetV(12.8)
+                .setVoltageRegulatorOn(true)
+                .setRegulatingTerminal(network.getLoad("B6-L").getTerminal())
+                .add();
+        parameters.setUseReactiveLimits(false);
+        parametersExt.setSecondaryVoltageControl(true);
+        network.newExtension(SecondaryVoltageControlAdder.class)
+                .newControlZone()
+                .withName("z1")
+                .newPilotPoint().withTargetV(13).withBusbarSectionsOrBusesIds(List.of("B10")).add()
+                .newControlUnit().withId("B5-G").add() // this control unit is a generator with remote voltage control
+                .newControlUnit().withId("B8-G").add()
+                .add()
+                .add();
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertEquals(LoadFlowResult.ComponentResult.Status.CONVERGED, result.getComponentResults().get(0).getStatus());
+        assertEquals(8, result.getComponentResults().get(0).getIterationCount());
+
+        assertVoltageEquals(13, b10);
+        assertVoltageEquals(12.682, b6);
+        assertVoltageEquals(25.311, b8);
+        assertReactivePowerEquals(82.904, g5.getTerminal());
+        assertReactivePowerEquals(-97, g8.getTerminal());
+    }
+
+    @Test
+    void testWithDiscardedGenerator() {
```

Comment:
This tests works also in main. Is it what you expect ? Or is it supposed to check something you wanted to fix ?